source/synthDrivers/sapi5.py

```diff
@@ -58,36 +86,127 @@ class SpeechVoiceEvents(IntEnum):
+	STRING = 4
+
+
+# Function types for calling COM methods
+_Com_AddRef = WINFUNCTYPE(c_ulong)(1, "AddRef")
+_Com_Release = WINFUNCTYPE(c_ulong)(2, "Release")
+_ISpEventSource_GetEvents = WINFUNCTYPE(HRESULT, c_ulong, POINTER(SPEVENT), POINTER(c_ulong))(11, "GetEvents")
+
+SPDFID_WaveFormatEx = GUID("{c31adbae-527f-4ff5-a230-f62bb61ff70c}")
+
+
+class _SapiEvent(SPEVENT):
+	def clear(self) -> None:
+		"""Clear and free related data."""
+		if self.elParamType in (_SPEventLParamType.TOKEN, _SPEventLParamType.OBJECT):
+			_Com_Release(cast(self.lParam, c_void_p))
+		elif self.elParamType in (_SPEventLParamType.POINTER, _SPEventLParamType.STRING):
+			windll.ole32.CoTaskMemFree(cast(self.lParam, c_void_p))
+		memset(byref(self), 0, sizeof(self))
+
+	def __del__(self):
+		self.clear()
+
+	@staticmethod
+	def copy(dst: SPEVENT, src: SPEVENT) -> None:
+		memmove(byref(dst), byref(src), sizeof(src))
+		if not src.lParam:
+			return dst
+		if src.elParamType == _SPEventLParamType.POINTER:
+			dst.lParam = windll.ole32.CoTaskMemAlloc(src.wParam)
+			if not dst.lParam:
+				raise COMError(hresult.E_OUTOFMEMORY, "CoTaskMemAlloc failed", (None, None, None, None, None))
+			memmove(dst.lParam, src.lParam, src.wParam)
+		elif src.elParamType == _SPEventLParamType.STRING:
+			strbuf = create_unicode_buffer(cast(src.lParam, c_wchar_p).value)
+			bufsize = sizeof(strbuf)
+			dst.lParam = windll.ole32.CoTaskMemAlloc(bufsize)
+			if not dst.lParam:
+				raise COMError(hresult.E_OUTOFMEMORY, "CoTaskMemAlloc failed", (None, None, None, None, None))
+			memmove(dst.lParam, byref(strbuf), bufsize)
+		elif src.elParamType in (_SPEventLParamType.TOKEN, _SPEventLParamType.OBJECT):
+			_Com_AddRef(cast(src.lParam, c_void_p))
+
+	def copyTo(self, dst: SPEVENT) -> None:
+		_SapiEvent.copy(dst, self)
+
+	def copyFrom(self, src: SPEVENT) -> None:
+		_SapiEvent.copy(self, src)
+
+	def getFrom(self, eventSource: _Pointer[ISpEventSource]) -> bool:
+		"""Get one event from the event source and store it in this object.
+		Return False if there is no event."""
+		self.clear()
+		# Use the raw ctypes.WINFUNCTYPE to call ISpEventSource::GetEvents()
+		# instead of using comtypes, because we want to retrieve the event in-place
+		# rather than letting comtypes allocating a new SPEVENT structure.
+		hr = _ISpEventSource_GetEvents(eventSource, 1, byref(self), None)
+		return hr == hresult.S_OK
+
+	@staticmethod
+	def enumerateFrom(eventSource: _Pointer[ISpEventSource]) -> Generator["_SapiEvent", None, None]:
+		"""Enumerate all events in the event source."""
+		while True:
+			event = _SapiEvent()
+			if not event.getFrom(eventSource):
+				break
+			yield event
+
+	def getString(self) -> str:
+		"""Get the string parameter stored in lParam."""
+		if self.elParamType != _SPEventLParamType.STRING:
+			raise TypeError("The lParam of this event is not a string")
+		return cast(self.lParam, c_wchar_p).value
 
 
-class SynthDriverAudioStream(COMObject):
+class SynthDriverAudio(COMObject):
```

Comment:
Yes this would be an API breaking change. I think just keeping the current name is fine, or moving to a private name and retaining this alias.